plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -3657,6 +3675,35 @@ private static function check_locked_orders( $response, $data ) {
 		return wc_get_container()->get( Automattic\WooCommerce\Internal\Admin\Orders\EditLock::class )->check_locked_orders_ajax( $response, $data );
 	}
 
+	// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter.FoundAfterLastUsed
+	/**
+	 * Render a variation editor.
+	 *
+	 * NOTE! Do NOT remove the apparently unused function arguments.
+	 * These are actually used inside the included html-variation-admin template.
+	 *
+	 * @param WC_Product $product_object PArent product of the variation being edited.
+	 * @param WC_Product $variation_object Variation being edited.
+	 * @param int        $loop Index of the variation being rendered.
+	 * @param float|null $base_cost Default cost for variations, null if the Cost of Goods Sold feature is disabled.
+	 */
+	private static function render_variation_html( WC_Product $product_object, WC_Product $variation_object, $loop, ?float $base_cost ) {
+		$variation_id   = $variation_object->save();
```

Comment:
I think it's great we're extracting mostly identical code to this function but why would we need to save the variation object to render some details about it?

While this is true for the AJAX callback for `add_variation` above ([ref](https://github.com/woocommerce/woocommerce/blob/5715bbd2f253eb3a43f3e60c0873ee9c09a70d0c/plugins/woocommerce/includes/class-wc-ajax.php#L837)), it's not true for `load_variations()` (see https://github.com/woocommerce/woocommerce/blob/5715bbd2f253eb3a43f3e60c0873ee9c09a70d0c/plugins/woocommerce/includes/class-wc-ajax.php#L2416).

Even if it ends up being innocuous, I'd prefer if we don't unnecessarily call `save()`.

Reply:
Good catch. Modified in [`51aee76` (#54399)](https://github.com/woocommerce/woocommerce/pull/54399/commits/51aee76b53e7a9fbcd066eb69cfcc07a948cfb6d) so that the saving happens externally to the method and only when needed.